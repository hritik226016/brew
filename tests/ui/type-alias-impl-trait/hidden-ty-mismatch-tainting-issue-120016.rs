//@ compile-flags: --edition=2021

#![feature(type_alias_impl_trait, const_async_blocks)]

struct Bug {
    V1: [(); {
        type F = impl std::future::Future<Output = impl Sized>;
        fn concrete_use() -> F {
            async {}
        }
        let f: F = async { 1 };
        //~^ ERROR concrete type differs from previous defining opaque type use
        //~| ERROR concrete type differs from previous defining opaque type use
        1
    }],
}

fn main() {}
