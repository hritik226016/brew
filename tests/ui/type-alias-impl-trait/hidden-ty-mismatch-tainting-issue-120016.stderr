error: concrete type differs from previous defining opaque type use
  --> $DIR/hidden-ty-mismatch-tainting-issue-120016.rs:11:20
   |
LL |         let f: F = async { 1 };
   |                    ^^^^^^^^^^^ expected `{async block@$DIR/hidden-ty-mismatch-tainting-issue-120016.rs:9:13: 9:18}`, got `{async block@$DIR/hidden-ty-mismatch-tainting-issue-120016.rs:11:20: 11:25}`
   |
note: previous use here
  --> $DIR/hidden-ty-mismatch-tainting-issue-120016.rs:9:13
   |
LL |             async {}
   |             ^^^^^^^^

error: concrete type differs from previous defining opaque type use
  --> $DIR/hidden-ty-mismatch-tainting-issue-120016.rs:11:20
   |
LL |         let f: F = async { 1 };
   |                    ^^^^^^^^^^^ expected `()`, got `i32`
   |
note: previous use here
  --> $DIR/hidden-ty-mismatch-tainting-issue-120016.rs:9:13
   |
LL |             async {}
   |             ^^^^^^^^

error: aborting due to 2 previous errors

